### PR TITLE
Gh-2977: GitHub Actions has deprecated actions using Node 12

### PR DESCRIPTION
**Pull request by GCHQDeveloper314**
PR is a draft as this is not a complete fix yet. See issue for why.

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -43,7 +44,7 @@ jobs:
 
     - name: Output branch name
       id: branch
-      run: echo "::set-output name=branch_name::$BRANCH_NAME"
+      run: echo "{branch_name}={$BRANCH_NAME}" >> $GITHUB_OUTPUT
```
Comment (t92549) [inline .github/workflows/release.yaml:47]: Haas this been tested on a fork or another workflow?

File: .github/workflows/create-release-branch.yaml [hunk 2]
```diff
@@ -36,29 +36,12 @@ jobs:
...
-      uses: actions-ecosystem/action-bump-semver@v1
-      id: bump-semver
-      with:
-        current_version: ${{ env.CURRENT_VERSION }}
-        level: minor
-
-    - name: Set version automatically
-      if: ${{ !github.event.inputs.version }}
-      run: echo "RELEASE_VERSION=$(echo ${{ steps.bump-semver.outputs.new_version }} )" >> $GITHUB_ENV
+      run: echo "RELEASE_VERSION=$(echo ${{ env.CURRENT_VERSION }} | sed -r 's/([0-9]+)\.([0-9]+)\.([0-9]+)/echo \1.$((\2+1)).\3/' | sh)" >> $GITHUB_ENV
```
Comment (t92549) [inline .github/workflows/create-release-branch.yaml:41]: Does this sed command bump the minor version or patch version? I noticed it's the same as the hotfix command but the hotfix is meant to bump patch (2.0.0 -> 2.0.1) and release is meant to bump minor (2.0.0 -> 2.1.0)

**Review by t92549 — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yaml (hunk 1 above)
Comment (GCHQDeveloper314) [reply .github/workflows/release.yaml:47]: No, but it is taken directly [from the GitHub docs for migrating](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/) away from `::set-output` which no longer works.

File: .github/workflows/create-release-branch.yaml (hunk 2 above)
Comment (GCHQDeveloper314) [reply .github/workflows/create-release-branch.yaml:41]: Yes this is a mistake, forgot to change to be the patch. I'll add a commit fixing this.

**Review by t92549 — APPROVED** [review]

**Pull request merged**